Examples/Algorithms/HepMC/include/ActsExamples/HepMC/EventExtraction.hpp

```diff
@@ -0,0 +1,55 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <memory>
+
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "Acts/Utilities/Units.hpp"
+#include "ActsExamples/EventData/SimParticle.hpp"
+#include "ActsExamples/Framework/BareAlgorithm.hpp"
+#include "ActsExamples/Framework/ProcessCode.hpp"
+
+class G4RunManager;
+
+namespace ActsExamples {
+
+/// @brief This class extracts a certain process from a HepMC event record.
+class EventExtraction final : public ActsExamples::BareAlgorithm {
+ public:
+  /// @class Config
+  struct Config {
+    /// The input collection
+    std::string inputEvents;
+    /// The output collection
+    std::string outputEventFraction = "event-fraction";
+    /// The process that should be extracted
+    std::string extractionProcess;
+
+    /// Minimum absolute value of considered PDG IDs
+    int minAbsPdg = 40;
+    /// Maximum absolute value of considered PDG IDs
+    int maxAbsPdg = 2212;
```

Comment:
Nitpicking: cut settings are typically named `<value>{Min,Max}` (see e.g. `ParticleSelector` configuration). I would suggest to use the same scheme here.

Reply:
done